### ISSUE TITLE
FR: Lint Dummy URLs to end with ".md"

### ISSUE THREAD

**Issue by linguaRoo** [platers/obsidian-linter#1363]
## Dummy Links Do Not Generate MD File Extension when Followed

When making dummy links in Obsidian, I often ignore the file extension (.md) when creating the link, particularly with Markdown links, and often forget to update them when sharing the Vault via GitHub or other services -- which means that other users outside of Obsidian cannot follow my links even when the pages are there, because while Obsidian assumes unmarked file extensions are *.md, other services make no such assumption.

## Check Links for Extensionless Target Path

A linting rule to ensure that links end with FULL-STOP (.) delimited file extensions. All it should need to do is check the string inside of the link target () and ensure that it contains a FULL STOP character (.) followed by one or more other characters. If no extension is detected, append ".md" to the end of the link target text.

### Possible Issues

This may fail on file naming conventions which use FULL STOP delimited words -- such.as.this -- which some people may use rather than underscores, dashes, or natural spacing conventions for longer filenames. A list of specific valid extensions might prevent this while still esnuring that links end with at least some valid extension?

## Intended Behavior:

``` markdown
- [link](path/to/link) ==> [link](path/to/link.md)
- [link](path/to/link.md) ==> _NO CHANGE_
- [link](path/to/link.htm) ==> _NO CHANGE_
```

## Considered Options

I've looked at using other plugins like Text Convert and scanned over the rest of the community plugins library and none seem to specifically address this particular issue, perhaps because Obsidian makes it a non-issue while remaining internal. None of the link formatters or converters I've seen seem to address this, and making this a linting feature seems useful for anybody who might find themselves with file links lacking extensions.

The alternative would be to rename the files, forcing Obsidian to update their links, but renaming them twice -- once to force an update, then again to the proper filename (which they already have) -- seems both tedious and cumbersome.

## Additional Context

The issue was brought up recently because I share constructed language documentation produced in Obsidian through GitHub, and was informed by an observer that many of my links to words do not work, because "the pages don't exist." After checking the links and Vault, I realized the issue was that because Obsidian assumes unmarked links to point to a *.md file, it takes no action to fix underdeveloped link targets -- it just assumes that a link to `file` is really a link to `file.md` if no extension is provided. As I often use the unmarked structure when creating dummy links in my documentation, and these links are not automatically updated when the file is created, I have a great many links across my conlang projects which would require a lot of time renaming files slightly, and renaming them back, to fix all of these issues.
